app/assets/javascripts/uploadcare/settings.coffee

```diff
@@ -201,7 +203,18 @@ uploadcare.namespace 'settings', (ns) ->
 
     if settings.validators
       settings.validators = settings.validators.slice()
-
+    
+    if settings.previewProxy and not settings.previewUrlCallback
+      settings.previewUrlCallback = (url, info) =>
+        encodedUrl = encodeURIComponent(url)
+        justAppend = /\=$/.test(settings.previewProxy)
+        useAmpersand = /[^\&\?]$/.test(settings.previewProxy)
```

Comment:
You shouldn't worry about existing ampersand here, because of `//base.com/?&url=`, `//base.com/?param=value&url=` and `//base.com/?param=value&&url=` — all valid URLs. Just always use `&url=` in this case.

But, there is another case: when no `?` in previewProxy at all. You should add it in this case:
`settings.previewProxy` + `?` + `&url=#{encodedUrl}`

Reply:
@homm `?&' and `&&` is weird despite that it is valid in terms of URL. Furthermore we can't know what URL parser will be used on the user's backend. Actually I've seen some old URL parser that don't comply spec and just can't parse such URL.

I didn't even consider that user will pass URL without `?`. Do we really need such flexibility?